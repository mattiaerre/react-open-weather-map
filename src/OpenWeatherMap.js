import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const OpenWeatherMap = ({ data, config }) => {
  if (data === OpenWeatherMap.defaultProps.data) {
    return (
      <div className={config.containerClassName} />
    );
  }

  const day = moment.unix(data.dt);
  const src = `//openweathermap.org/img/w/${data.weather[0].icon}.png`;

  return (
    <div className={config.containerClassName}>
      <div className="main-heading">{data.name}, {data.sys.country}</div>
      <div className="day">{day.format('dddd h:mm A')}</div>
      <div className="description">{data.weather[0].description}</div>
      <div className="icon">
        <img src={src} alt={data.weather[0].description} />
      </div>
      <div className="temperature">{data.main.temp} &deg;C</div>
    </div>
  );
};

OpenWeatherMap.propTypes = {
  data: PropTypes.object,
  config: PropTypes.object,
};

OpenWeatherMap.defaultProps = {
  data: { weather: [{}], sys: {}, main: {} },
  config: { containerClassName: 'react-open-weather-map' },
};

export default OpenWeatherMap;
