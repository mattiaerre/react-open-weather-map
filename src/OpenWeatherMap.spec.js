import React from 'react';
import { create } from 'react-test-renderer';
import OpenWeatherMap from './OpenWeatherMap';
import sample from '../data/current-weather-data-sample.json';

describe('<OpenWeatherMap />', () => {
  const data = sample.weather;
  const config = { containerClassName: 'container-class-name' };

  const scenarios = [
    { description: 'w/o data w/o config', props: {} },
    { description: 'w data w/o config', props: { data } },
    { description: 'w/o data w config', props: { config } },
    { description: 'w data w config', props: { data, config } }
  ];

  scenarios.forEach((scenario) => {
    test(scenario.description, () => {
      const component = create(<OpenWeatherMap {...scenario.props} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
