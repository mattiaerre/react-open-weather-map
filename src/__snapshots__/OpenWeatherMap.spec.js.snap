// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpenWeatherMap /> w data w config 1`] = `
<div
  className="container-class-name"
>
  <div
    className="main-heading"
  >
    London
    , 
    GB
  </div>
  <div
    className="day"
  >
    Saturday 2:07 PM
  </div>
  <div
    className="description"
  >
    light rain
  </div>
  <div
    className="icon"
  >
    <img
      alt="light rain"
      src="//openweathermap.org/img/w/10n.png"
    />
  </div>
  <div
    className="temperature"
  >
    16.62
     °C
  </div>
</div>
`;

exports[`<OpenWeatherMap /> w data w/o config 1`] = `
<div
  className="react-open-weather-map"
>
  <div
    className="main-heading"
  >
    London
    , 
    GB
  </div>
  <div
    className="day"
  >
    Saturday 2:07 PM
  </div>
  <div
    className="description"
  >
    light rain
  </div>
  <div
    className="icon"
  >
    <img
      alt="light rain"
      src="//openweathermap.org/img/w/10n.png"
    />
  </div>
  <div
    className="temperature"
  >
    16.62
     °C
  </div>
</div>
`;

exports[`<OpenWeatherMap /> w/o data w config 1`] = `
<div
  className="container-class-name"
/>
`;

exports[`<OpenWeatherMap /> w/o data w/o config 1`] = `
<div
  className="react-open-weather-map"
/>
`;
